src/pam_unix_ng-acct.c

```diff
@@ -38,6 +44,22 @@ acct_mgmt(pam_handle_t *pamh, struct config_t *cfg)
       pam_syslog(pamh, LOG_ERR, "pwaccess expired failed: %s",
 		 error ? error : strerror(-r));
 
+      // The body of this `if` block could be candidate for placing it into a
+      // dedicated function, since it is performing a very specific fallback
+      // purpose.
```

Comment:
Yes, but then you have the problem that you need two return values.
In go this would be simple to solve...